pkg/common/resources/resources.go

```diff
@@ -44,6 +46,55 @@ func (q Quantity) string() string {
 	return strconv.FormatInt(int64(q), 10)
 }
 
+// Util struct to keep track of application resource usage
+type UsedResourceTracker struct {
+	// Two level map for aggregated resource usage
+	// With instance type being the top level key, the mapped value is a map:
+	//   resource type (CPU, memory etc) -> the aggregated used time (in seconds) of the resource type
+	//
+	UsedResourceMap map[string]map[string]int64
+
+	sync.RWMutex
+}
+
+// Aggregate the resource usage to UsedResourceMap[instType]
+// The time the given resource used is the delta between the resource createTime and currentTime
+func (urt *UsedResourceTracker) AggregateUsedResource(instType string,
+	resource *Resource, createTime time.Time) {
+	var aggregatedResourceTime map[string]int64
+	var timeDiff int64
+	var releaseTime time.Time
+	var resourceSecondsKey string
```

Comment:
Declare the variables close to their usage, increasing readability. Most of the time you can use `:=`.
Eg.
```
releaseTime := time.Now()
timeDiff := int64(releaseTime.Sub(createTime).Seconds())
resourceSecondsKey := key + "Seconds"
```
It's preferred to start the method with Lock/Unlock pair if there's no reason to do otherwise.

Reply:
Thanks for the review and very good feedback Pbacsko! Go language is new to me and I'm still learning, will address in next rev.